source/titback/interpreter.cpp

```diff
@@ -0,0 +1,141 @@
+  return config_.get();
+}
+
+void PythonConfig::set_home(const char* home) const {
+  TIT_ASSERT(home != nullptr, "Home directory must be specified!");
+  const auto status = PyConfig_SetBytesString(base(), &base()->home, home);
+  if (PyStatus_IsError(status) == 0) return;
+  TIT_THROW("Failed to set Python home directory to '{}': {}: {}.",
+            home,
+            status.func,
+            status.err_msg);
+}
+
+void PythonConfig::set_prog_name(const char* name) const {
+  TIT_ASSERT(name != nullptr, "Program name must be specified!");
+  const auto status =
+      PyConfig_SetBytesString(base(), &base()->program_name, name);
+  if (PyStatus_IsError(status) == 0) return;
+  TIT_THROW("Failed to set Python program name to '{}': {}: {}.",
+            name,
+            status.func,
+            status.err_msg);
+}
+
+void PythonConfig::set_cmd_args(CmdArgs args) const {
+  const auto status = PyConfig_SetBytesArgv(
+      base(),
+      static_cast<Py_ssize_t>(args.size()),
+      const_cast<char* const*>(args.data())); // NOLINT(*-const-cast)
+  if (PyStatus_IsError(status) == 0) return;
+  TIT_THROW("Failed to set Python command line arguments: {}: {}.",
+            status.func,
+            status.err_msg);
+}
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+PythonInterpreter::PythonInterpreter(PythonConfig config)
+    : config_{std::move(config)} {
+  // Initialize the Python interpreter.
+  const auto status = Py_InitializeFromConfig(config_.base());
+  if (PyStatus_IsError(status) != 0) {
+    TIT_THROW("Failed to initialize Python interpreter: {}: {}.",
+              status.func,
+              status.err_msg);
+  }
+
+  // Get the globals of the main module.
+  auto* const main_module = PyImport_AddModule("__main__");
+  if (main_module == nullptr) {
+    PyErr_Print();
+    TIT_THROW("Failed to import the main module.");
+  }
+  globals_ = PyModule_GetDict(main_module);
+  if (globals_ == nullptr) {
+    PyErr_Print();
+    TIT_THROW("Failed to get the main module globals.");
+  }
+}
+
+PythonInterpreter::~PythonInterpreter() {
+  Py_Finalize();
+}
+
+auto PythonInterpreter::exec(const char* statement) const -> bool {
+  TIT_ASSERT(statement != nullptr, "Python statement must be specified!");
+  auto* const result = PyRun_StringFlags(statement,
+                                         /*start=*/Py_file_input,
+                                         /*globals=*/globals_,
+                                         /*locals=*/globals_,
+                                         /*flags=*/nullptr);
+  if (result != nullptr) {
+    Py_DECREF(result);
+    return true;
+  }
+
+  PyErr_Print();
+  return false;
+}
+
+auto PythonInterpreter::exec_file(const char* file_name) const -> bool {
+  TIT_ASSERT(file_name != nullptr, "Python file name must be specified!");
+  auto* const file = std::fopen(file_name, "r"); // NOLINT(*-owning-memory)
```

Comment:
Let's make a wrapper based on `std::unique_ptr` in order to not distress Sonar and Clang Tidy.